uqcsbot/starboard.py

```diff
@@ -0,0 +1,202 @@
+import os
+from typing import Optional
+
+import discord
+from discord.ext import commands
+
+from uqcsbot import models
+# needs to be models and not just starboard because of namespacing with this class
+
+class Starboard(commands.Cog):
+    CHANNEL_NAME = "starboard"
+    EMOJI_NAME = "starhaj"
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+        self.base_threshold = int(os.environ.get("SB_BASE_THRESHOLD"))
+        self.big_threshold = int(os.environ.get("SB_BIG_THRESHOLD"))
+    
+    @commands.Cog.listener()
+    async def on_ready(self):
+        """ 
+        Really this should be in __init__ but this stuff doesn't exist until the bot is ready.
+        N.B. this does assume the bot only has access to one channel called "starboard" and one emoji called "neat". If this assumption stops holding, we may need to move back to IDs (cringe) or ensure we only get them from the correct guild (cringer).
```

Comment:
Nit: Emoji isn't called `neat` anymore

Reply:
Style nit: While we don't have a formal style, probably split this line so we can keep this line below 120 chars.

Reply:
Turns out I changed this locally and just didn't commit it :100: